crates/hotshot/src/lib.rs

```diff
@@ -444,78 +460,91 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
             internal_event_stream: internal_event_stream.clone(),
             hotshot: self.clone(),
             storage: self.inner.storage.clone(),
         };
 
-        let task_runner = add_network_message_task(
-            task_runner,
-            internal_event_stream.clone(),
-            quorum_network.clone(),
-        )
-        .await;
-        let task_runner = add_network_message_task(
-            task_runner,
-            internal_event_stream.clone(),
-            da_network.clone(),
-        )
-        .await;
+        add_network_message_task(registry.clone(), event_tx.clone(), quorum_network.clone()).await;
+        add_network_message_task(registry.clone(), event_tx.clone(), da_network.clone()).await;
 
-        let task_runner = add_network_event_task(
-            task_runner,
-            internal_event_stream.clone(),
+        add_network_event_task(
+            registry.clone(),
+            event_tx.clone(),
+            event_rx.activate_cloned(),
             quorum_network.clone(),
             quorum_membership,
-            NetworkTaskKind::Quorum,
+            network::quorum_filter,
         )
         .await;
-        let task_runner = add_network_event_task(
-            task_runner,
-            internal_event_stream.clone(),
+        add_network_event_task(
+            registry.clone(),
+            event_tx.clone(),
+            event_rx.activate_cloned(),
             da_network.clone(),
             da_membership,
-            NetworkTaskKind::Committee,
+            network::committee_filter,
         )
         .await;
-        let task_runner = add_network_event_task(
-            task_runner,
-            internal_event_stream.clone(),
+        add_network_event_task(
+            registry.clone(),
+            event_tx.clone(),
+            event_rx.activate_cloned(),
             quorum_network.clone(),
             view_sync_membership,
-            NetworkTaskKind::ViewSync,
+            network::view_sync_filter,
         )
         .await;
-        let task_runner = add_network_event_task(
-            task_runner,
-            internal_event_stream.clone(),
+        add_network_event_task(
+            registry.clone(),
+            event_tx.clone(),
+            event_rx.activate_cloned(),
             quorum_network.clone(),
             vid_membership,
-            NetworkTaskKind::VID,
+            network::vid_filter,
         )
         .await;
-        let task_runner = add_consensus_task(
-            task_runner,
-            internal_event_stream.clone(),
-            output_event_stream.clone(),
-            handle.clone(),
+        let consensus_state = add_consensus_task(output_event_stream.0.clone(), &handle).await;
+        let task = Task::new(
+            event_tx.clone(),
+            event_rx.activate_cloned(),
+            registry.clone(),
+            consensus_state,
+        );
+        registry.run_task(task).await;
```

Comment:
Related to my comment on `add_consensus_task`, I'm wondering why the consensus task is unique and we need to call `run_task` for it here but not other tasks.

Reply:
Yeah it's for the unit tests, the other task states are trivial to construct so for unit test I was just doing that, but for consensus I just reused the state creation for `test_consensus_task`

Reply:
I'll make them all match